### DISCUSSION TITLE
Is it possible to implement a database connection pool plugin?

### DISCUSSION THREAD

**Discussion by ryan053857** [roadrunner-server/roadrunner#2054] [Ideas]
php connects to golang through the rpc method provided by goridge, then processes the database operations in golang and returns the results to php, like redis kv plugin

**Comment by rustatian** [comment]
Hey @shellphy 👋 
Sure, pooling is native to Go. But you know, transactions, queries, all this stuff should work via RPC methods. But there are many cases which should be somehow solved, for example: if the worker failed, like, restarted for example while handling a transaction.
